pkg/authorization/workspace_content_authorizer.go

```diff
@@ -100,91 +101,89 @@ func (a *workspaceContentAuthorizer) Authorize(ctx context.Context, attr authori
 		&rbac.ClusterRoleBindingLister{Lister: parentWorkspaceKubeInformer.ClusterRoleBindings().Lister()},
 	)
 
-	// TODO: decide if we want to require workspaces for all kcp variations. For now, only check if the workspace controllers are running,
-	// as that ensures the ClusterWorkspace CRD is installed, and that our shared informer factory can sync all its caches successfully.
-	if a.clusterWorkspaceLister != nil {
-		// check the workspace even exists
-		// TODO: using scoping when available
-		if ws, err := a.clusterWorkspaceLister.Get(clusters.ToClusterAwareKey(parentClusterName, clusterWorkspace)); err != nil {
-			if errors.IsNotFound(err) {
-				return authorizer.DecisionDeny, fmt.Sprintf("%q workspace access not permitted", cluster.Name), nil
-			}
-			return authorizer.DecisionNoOpinion, "", err
-		} else if len(ws.Status.Initializers) > 0 {
+	extraGroups := sets.NewString()
+
+	// check the workspace even exists
+	if ws, err := a.clusterWorkspaceLister.Get(clusters.ToClusterAwareKey(parentClusterName, clusterWorkspace)); err != nil {
+		if errors.IsNotFound(err) {
+			return authorizer.DecisionDeny, fmt.Sprintf("%q workspace access not permitted", cluster.Name), nil
+		}
+		return authorizer.DecisionNoOpinion, "", err
```

Comment:
Is this the sort of place we want to bubble up infra level issues to the user? Why not a generic 500?